react-common/components/controls/EmbedVideo.tsx

```diff
@@ -1,45 +1,67 @@
-import * as React from "react";
+
 import { classList, ControlProps } from "../util";
+import { useRef, useEffect } from 'react';
+import * as React from 'react';
+import { MediaPlayer } from "dashjs";
 
 export interface EmbedVideoProps extends ControlProps {
-  src: "youtube";
-  id: string;
-  autoplay?: boolean;
-  allowFullScreen?: boolean;
+    src: "youtube" | "streams";
+    id: string;
+    autoplay?: boolean;
+    allowFullScreen?: boolean;
 }
 
-
-
 export const EmbedVideo = (props: EmbedVideoProps) => {
     const {
-       src,
-       id,
-       autoplay,
-       allowFullScreen
+        src,
+        id,
+        autoplay,
+        allowFullScreen
     } = props;
 
+    let videoURL: string;
 
+    if (src == "youtube") {
 
-    let videoURL = `https://www.youtube.com/embed/${id}?${autoplay ? "autoplay=1" : ""}`;
+        videoURL = `https://www.youtube.com/embed/${id}?${autoplay ? "autoplay=1" : ""}`;
+        let appending = "";
+        if (autoplay) {
+            appending = "; autoplay";
+        }
 
-    let appending = "";
+        let st1 = "encrypted-media; picture-in-picture";
+        if (pxt.BrowserUtils.isIOS()) {
+            st1 += "; muted=true";
+        }
+        let st2 = st1 + appending;
 
-    if (autoplay){
-        appending = "; autoplay";
-    }
+        return (
+            <div className={classList("common-video")}>
+                <iframe src={videoURL} title="YouTube video player"
+                    frameBorder="0" allow={st2} allowFullScreen={allowFullScreen}></iframe>
+            </div>
+        );
 
-    let st1 = "encrypted-media; picture-in-picture";
+    } else {
 
-    if (pxt.BrowserUtils.isIOS()){
-        st1 += "; muted=true";
-    }
+        videoURL = `https://makecode-lucas-testing-makecodetempmediaservice-usea.streaming.media.azure.net/${id}/manifest(format=mpd-time-cmaf)`;
```

Comment:
This shouldn't be checked-in?

Reply:
This can change between various subscription or we create a new resources, This needs to be inferred. Similar to immersive reader.

Reply:
fair point, i wonder if another parameter might have to get passed in to account for the specific streaming endpoint as well as locator will be used